[Draft] Move specification from ROS wiki to the urdfdom repo (#183)
ros/urdfdom: specification/robot.md

# `<robot>` element

> The root element in a robot description file must be a `<robot>` element, with all other elements must be encapsulated within.

Reviewer: This is another place where I think we can make the update to the language in this PR.

## Attributes

| `<robot>` attr | type   | use      | default value | description                                                                                                                                                                                                               |
| -------------- | ------ | -------- | ------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | `name`         | string | required | NA            | The master file must have a *name* attribute. The *name* attribute is optional in included files. If the attribute *name* is specified in an additional included file, it must have the same value as in the master file. |

Reviewer: This is another place where I think we should update the language.  As @peci1 points out, this doesn't actually make too much sense currently.

> | `name`         | string | required | NA            | The master file must have a *name* attribute. The *name* attribute is optional in included files. If the attribute *name* is specified in an additional included file, it must have the same value as in the master file. |

Reviewer 2: I agree

> | `name`         | string | required | NA            | The master file must have a *name* attribute. The *name* attribute is optional in included files. If the attribute *name* is specified in an additional included file, it must have the same value as in the master file. |

Author: ```suggestion
| `name`         | string | required | NA            | The `robot` element must have a *name* attribute. |
```

| `version`      | string | optional | "1.0"         | version of the urdf specification in `<major>.<minor>` format.                                                                                                                                                            |

## Elements

| `<robot>` element                     | description                                                          |
| ------------------------------------- | -------------------------------------------------------------------- |
| [`<link>`](./link.md)                 | defines a link with its own frame.                                   |
| [`<joint>`](./joint.md)               | mandatory joint frame definition.                                    |

> | [`<transmission>`](./transmission.md) | (PR2 specific).                                                      |

Reviewer: This references `transmission.md`, but there is no file here with that name.

Reviewer 2: Do you think it makes more sense to remove it? `<transmission>` is not part of the parser and is specific to PR2 and `ros_control` so it doesn't seem to be part of the specification. Or what do you recommend?

Author: Good point, that was an error on my side. @sloretz @scpeters do you prefer that I import http://wiki.ros.org/urdf/XML/Transmission as well here, or remove it?

Reviewer 3: I'd recommend removing it for now. We can decide how to handle extensions later.

> | [`<gazebo>`](./gazebo.md)             | [Gazebo](http://wiki.ros.org/gazebo_ros_pkgs) simulation extensions. |

Reviewer: Add `<sensor>` to this list.

Author: ```suggestion
| [`<sensor>`](./sensor.md)             | Sensor extensions. |
| [`<gazebo>`](./gazebo.md)             | [Gazebo](http://wiki.ros.org/gazebo_ros_pkgs) simulation extensions. |
```
